clients/drcachesim/tracer/func_trace.cpp

```diff
@@ -104,13 +104,17 @@ func_pre_hook(void *wrapcxt, INOUT void **user_data)
     size_t idx = (size_t)*user_data;
     func_metadata_t *f = (func_metadata_t *)drvector_get_entry(&funcs, (uint)idx);
     app_pc retaddr = drwrap_get_retaddr(wrapcxt);
-    append_entry(drcontext, TRACE_MARKER_TYPE_FUNC_ID, (uintptr_t)f->id);
-    append_entry(drcontext, TRACE_MARKER_TYPE_FUNC_RETADDR, (uintptr_t)retaddr);
+    if (append_entry != NULL) {
```

Comment:
Hmm, it seems like append_entry is integral to the operation of this function tracing feature, and is not optional.  I would expect func_trace_init to fail if it's passed NULL instead of silently allowing it?  Unless you want this feature of instrumenting but not recording for testing/performance analysis purposes?  But then I'd expect some warning or comment or sthg.